### ISSUE TITLE
[BUG] Plugin incorrectly labeling valid class as invalid

### ISSUE THREAD

**Issue by zorrodg** [sergioazoc/oxlint-tailwindcss#37]
Hello! Thank you for your contribution to the community with this project.

Attaching a description of the bug in VSCode (although the CLI reporter shows it too).

<img width="893" height="173" alt="Image" src="https://github.com/user-attachments/assets/e2abe737-d15d-4f30-9457-fec28fc13b14" />

Here are the [Tailwind docs](https://tailwindcss.com/docs/responsive-design#using-size-containers) that mention the class. Please let me know if you need me to add further information.

Thanks!

**Comment by sergioazoc** [comment]
Hi @zorrodg — thanks for the report, this was a genuine bug and it's now fixed in **v1.3.1** (just
published to npm).

**Root cause:** `@container-size` is a valid Tailwind v4 utility (`container-type: size`), but the
plugin builds its set of valid classes by filtering Tailwind's `getClassList()`, and
`getClassList()` happens to omit `@container-size` (it lists `@container` and `@container-normal`,
but not this one). So `no-unknown-classes` treated it as unknown and offered the nearest Levenshtein
neighbor — `contain-size`.

Your report prompted a full audit of that class of gap (Tailwind's utility registry vs
`getClassList()`), which surfaced — and fixed — a handful of siblings in the same situation:

- `filter-none` / `backdrop-filter-none` and `max-w-screen`
- negative utilities Tailwind omits from `getClassList()`: `-hue-rotate-45`, `-col-1`, `-row-2`,
  `-backdrop-hue-rotate-30`
- the v3 background/object-position spellings (`bg-left-top`, `object-right-bottom`, …) —
  `enforce-canonical` now also rewrites these to the v4 form (`bg-top-left`, `object-bottom-right`,
  …)

Every addition is validated through Tailwind's own `candidatesToCss()`, so it stays correct across
Tailwind versions.

To pick up the fix:

```sh
npm i -D oxlint-tailwindcss@latest
```

The design-system cache invalidates automatically on the upgrade; if you're using the editor
integration (e.g. VS Code), restart the oxlint server or reload the window so it loads the new
plugin version. Let me know if you still see it — thanks again for the clear repro and the
screenshot! 🙏

**Comment by andreszorro** [comment]
Thank you for your prompt response! Will update.
